src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -76,20 +84,49 @@ public EvaluationContext getEvaluationContext() {
      * {@inheritDoc}
      */
     public void setProvider(FeatureProvider provider) {
+        if (provider == null) {
+            this.providers.remove(DEFAULT_PROVIDER_KEY);
+        } else {
+            setProvider(DEFAULT_PROVIDER_KEY, provider);
+        }
+    }
+
+    /**
+     * Add a provider for a named client.
+     * @param clientName The name of the client.
+     * @param provider The provider to set.
+     */
+    public void setProvider(String clientName, FeatureProvider provider) {
         try (AutoCloseableLock __ = providerLock.writeLockAutoCloseable()) {
-            this.provider = provider;
+            this.providers.put(clientName, provider);
         }
     }
 
     /**
      * {@inheritDoc}
      */
     public FeatureProvider getProvider() {
+        return this.providers.get(DEFAULT_PROVIDER_KEY);
+    }
+
+    /**
+     * Fetch a provider for a named client. If not found, return the default.
+     * @param name The client name to look for.
+     * @return A named {@link FeatureProvider}
+     */
+    public FeatureProvider getProvider(String name) {
         try (AutoCloseableLock __ = providerLock.readLockAutoCloseable()) {
-            return this.provider;
+            if (name != null) { // this happens when a client name isn't set.
+                FeatureProvider val = this.providers.get(name);
+                if (val != null) {
+                    return val;
+                }
+            }
+            return this.providers.get(DEFAULT_PROVIDER_KEY);
         }
     }
 
+
```

Comment:
don't know, whether u would like to remove the `inheritDoc` lines below in this PR. as the class is neither derived (except for java.lang.Object) nor implementing an interface, it can't inherit docs (on that method)...

Reply:
Tracked here: https://github.com/open-feature/java-sdk/issues/424